code/modules/spells/spell_types/shapeshift/_shape_status.dm

```diff
@@ -31,9 +31,10 @@
 
 /datum/status_effect/shapechange_mob/on_apply()
 	caster_mob.mind?.transfer_to(owner)
-	caster_mob.forceMove(owner)
+	caster_mob.moveToNullspace()
 	ADD_TRAIT(caster_mob, TRAIT_NO_TRANSFORM, TRAIT_STATUS_EFFECT(id))
 	caster_mob.apply_status_effect(/datum/status_effect/grouped/stasis, STASIS_SHAPECHANGE_EFFECT)
+	caster_mob.ai_controller = null
```

Comment:
I think you probably also have to delete it, this runtimes when I transform back from being a spider because the spider's AI controller notices that it is in nullspace
By the time you null it the controller already has a reference to the mob, is inside several lists, and doesn't need a reference back in order to continue existing